src/pal/src/file/file.cpp

```diff
@@ -1716,8 +1776,12 @@ GetFileAttributesW(
         goto done;
     }
     
-    size = WideCharToMultiByte( CP_ACP, 0, lpFileName, -1, filename, MAX_LONGPATH,
+    length = (PAL_wcslen(lpFileName)+1) * MaxWCharToAcpLengthFactor;
```

Comment:
@Priya91, there are quite a few places in this change that have very similar pattern: get length, open buffer, call WideCharToMultiByte, close buffer, etc. Do you think it might be useful to add a helper function that does all these operations in one place? Thank you.